File: easy/hidden_digits.lua
for line in io.lines(arg[1]) do
  line = line:gsub("[^0-9a-j]", "")
  for i in line:gmatch(".") do
    io.write(i:match("[a-j]") and i:byte()-string.byte("a") or i)
  end
  print((#line > 0) and "" or "NONE")
end

File: easy/hidden_digits.rb
h = { 'a' => 0, 'b' => 1, 'c' => 2, 'd' => 3, 'e' => 4,
      'f' => 5, 'g' => 6, 'h' => 7, 'i' => 8, 'j' => 9 }
File.open(ARGV[0]).each_line do |line|
  s = line.gsub(/[^\da-j]/, '').gsub(/[a-j]/, h)
  puts s.length > 0 ? s : 'NONE'
end
